build_runner/lib/src/generate/watch_impl.dart

```diff
@@ -240,6 +240,10 @@ class WatchImpl implements BuildState {
             .hasBeenUpdated(mergedChanges.keys.toSet())) {
           _terminateCompleter.complete();
           _logger.severe('Terminating builds due to build script update');
+
+          /// Intentional unhandled async error here, which will be caught at the
```

Comment:
[nit] double slash comment

I'm a little uncertain about this trick... I wonder if it would be more explicit to have a global variable through which we can report the status?